src/qilib/utils/storage/mongo.py

```diff
@@ -71,12 +73,17 @@ def __init__(self, name: str, host: str = 'localhost', port: int = 27017, databa
             host: MongoDB host
             port: MongoDB port
             database: The database to use, if empty the name of the storage is used
+            connection_timeout: How long to try to connect to database before raising an error
+        Raises:
+            StorageTimeoutError: If connection to database has not been established before connection_timeout is reached
         """
         super().__init__(name)
 
         type_registry = TypeRegistry([NumpyArrayCodec()])
         codec_options = CodecOptions(type_registry=type_registry)
-        self._client = MongoClient(host, port)
+        self._client = MongoClient(host, port, serverSelectionTimeoutMS=connection_timeout)
```

Comment:
I think you would also have to configure connectionTimeoutMS. And increase the serverSelectionTimeoutMS to make it 30 seconds.
--Copied from  from mongo_client.py ---
`serverSelectionTimeoutMS`: (integer) Controls how long (in milliseconds) the driver will wait to find an available, appropriate server to carry out a database operation; while it is waiting, multiple server monitoring operations may be carried out, each controlled by `connectTimeoutMS`. Defaults to ``30000`` (30 seconds).

Reply:
I think just using connectTimeoutMS should be enough, and I did change it to use it instead of the serverSelectionTimeoutMS

Reply:
I don's see `connectTimeoutMS` having any effect on the initial `_check_server_connection` timeout set by `serverSelectionTimeoutMS`:
```
import time

from qilib.utils.storage import StorageMongoDb
from qilib.utils.storage.interface import StorageTimeoutError

start = time.time()
try:
    storage = StorageMongoDb('dem-1199', port=42, connection_timeout=60000)
except StorageTimeoutError:
    pass
end = time.time()

print(f'it took {end - start} to construct storage')
```
This times out after 60 seconds